crates/hdk_derive/src/dna_properties.rs

```diff
@@ -0,0 +1,29 @@
+use proc_macro::TokenStream;
+use proc_macro_error::abort;
+use syn::parse_macro_input;
+use syn::Item;
+use syn::ItemStruct;
+
+pub fn build(_attrs: TokenStream, input: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(input as Item);
+    let ident = match &input {
+        Item::Struct(ItemStruct { ident, .. }) => ident,
+        _ => abort!(input, "dna_properties macro can only be used on Structs"),
+    };
+
+    let output = quote::quote! {
+        use std::any::type_name;
+        #[derive(Serialize, Deserialize, SerializedBytes, Debug)]
+        #input
+
+        trait TryFromDnaProperties {
+            fn try_from_dna_properties() -> ExternResult<#ident> {
+                #ident::try_from(dna_info()?.modifiers.properties)
+                    .map_err(|_| wasm_error!(WasmErrorInner::Guest(format!("Failed to deserialize DNA properties into {:}", type_name::<#ident>()))))
+            }
+        }
+
+        impl TryFromDnaProperties for #ident {}
+    };
```

Comment:
This creates a trait with a default implementation for the specific type and then writes an empty implementation for that type. Can the trait be generic outside of the macro and the macro implement it for the specific type? That sounds more explicit in my mind, but I don't know if that works.

Reply:
right yeah that does seem wrong -- okay moved the trait definition into `holochain_integrity_types` without any default implementation